pointcloud/basic_octree.go

```diff
@@ -182,3 +284,71 @@ func (octree *BasicOctree) CollidesWithGeometry(geom spatialmath.Geometry, thres
 	}
 	return false, errors.New("unknown octree node type")
 }
+
+// CollidesWith checks if the given octree collides with the given geometry and returns true if it does.
+func (octree *BasicOctree) CollidesWith(geom spatialmath.Geometry) (bool, error) {
```

Comment:
Whats the thought process of keeping this separate from `CollidesWithGeometry`?

Reply:
This preserves the ability to user-specify a confidence threshold and a collision buffer; I like keeping them separate